### PR TITLE
Fix setState error in Tip

### PR DESCRIPTION
**Pull request by zpfled**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "presets": ["es2015", "react"],
+  "presets": ["es2015", "react", "stage-1"],
```
Comment (zpfled) [inline .babelrc]: allow for use of decorators

File: components/Tip.js [hunk 2]
```diff
@@ -13,18 +14,17 @@ class Tip extends Component {
...
   }
 
   componentWillUnmount() {
-    window.removeEventListener('scroll', this.handleScroll);
-  }
-
-  handleScroll() {
-    _.debounce(() => this.setPlacement(), 200);
+    window.removeEventListener('scroll', this._setPlacement);
+    this.setPlacement.clear()
```
Comment (zpfled) [inline components/Tip.js]: clears delayed setPlacement calls

File: components/Tip.js [hunk 3]
```diff
@@ -13,18 +14,17 @@ class Tip extends Component {
...
   }
 
+  @Debounce(200)
```
Comment (zpfled) [inline components/Tip.js]: fancy

**Comment by zpfled** [comment]
@moodysalem

File: components/Tip.js (hunk 2 above)
Comment (moodysalem) [inline components/Tip.js]: Incorrect function name, this._setPlacement

Comment (zpfled) [inline components/Tip.js]: arg, thanks

File: package.json [hunk 4]
```diff
@@ -45,12 +45,15 @@
...
     "babel-loader": "^6.2.4",
+    "babel-plugin-transform-decorators-legacy": "^1.3.4",
     "babel-plugin-transform-require-ignore": "0.0.2",
     "babel-preset-es2015": "^6.6.0",
     "babel-preset-react": "^6.5.0",
+    "babel-preset-stage-1": "^6.5.0",
     "babel-register": "^6.7.2",
     "cross-env": "^1.0.7",
     "css-loader": "^0.23.1",
+    "debounce-decorator": "^1.0.4",
```
Comment (zpfled) [inline package.json]: Here @moodysalem

File: components/Tip.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ import ReactDOM from 'react-dom';
 import PureRenderMixin from 'react-addons-pure-render-mixin';
 import './../styles/components/Tip.scss';
 import _ from 'underscore';
+import Debounce from 'debounce-decorator';
```
Comment (zpfled) [inline components/Tip.js]: and here @moodysalem

File: components/Tip.js [hunk 6]
```diff
@@ -20,7 +20,7 @@ class Tip extends Component {
 
   componentWillUnmount() {
     window.removeEventListener('scroll', this._setPlacement);
-    this.setPlacement.clear()
+    this._setPlacement.clear()
```
Comment (moodysalem) [inline components/Tip.js]: actually this is wrong-I didn't realize this._setPlacement was defined the first time i looked at the PR and my comment was invalid

Comment (moodysalem) [inline components/Tip.js]: I don't think this._setPlacement is going to have a clear method because that is not the debounced function

**Comment by moodysalem** [comment]
Remove line 
Tip.mixins = [ PureRenderMixin ];

**Comment by moodysalem** [comment]
Bind to scroll event of scrollParent

Scroll events do not bubble AFAIK
https://developer.mozilla.org/en-US/docs/Web/Events/scroll

**Comment by zpfled** [comment]
I gotta leave this alone for now. I fixed the debounce bug and refined the placement generator, but the scroll parent binding isn't working and I need to work on the delete. I think this is a problem that will be pretty rare....we can loop back to it later when we have extra time.

**Pull request merged**